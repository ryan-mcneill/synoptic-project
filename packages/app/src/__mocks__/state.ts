export const mockState = {
  albums: {
    loading: false,
    data: {
      "5f1423c54ee369215e05625e": {
        name: "Sample Album A",
        songs: [
          {
            _id: "5f14218d4ee369215e05624f",
            name: "Hey",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056256",
            name: "Cute",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056257",
            name: "Going Higher",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      },
      "5f1423c54ee369215e05625f": {
        name: "Sample Album B",
        songs: [
          {
            _id: "5f1422054ee369215e056255",
            name: "A New Beginning",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          }
        ]
      }
    },
    error: null
  },
  artists: {
    loading: false,
    data: {
      "5f14228b4ee369215e05625a": {
        name: "Benjamin Tissot",
        songs: [
          {
            _id: "5f14218d4ee369215e05624f",
            name: "Hey",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056256",
            name: "Cute",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056257",
            name: "Going Higher",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      },
      "5f14228b4ee369215e05625b": {
        name: "BenSound",
        songs: [
          {
            _id: "5f1422054ee369215e056255",
            name: "A New Beginning",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      }
    },
    error: null
  },
  currentSong: {
    isPlaying: false,
    background: {
      isDark: false,
      colour: "#e0c1e2"
    }
  },
  menu: {
    isOpen: false,
    selectedTab: "PLAYLIST"
  },
  playlist: {
    previous: [],
    current: {
      _id: "5f14218d4ee369215e05624f",
      name: "Hey",
      artists: [
        {
          _id: "5f14228b4ee369215e05625a",
          name: "Benjamin Tissot"
        }
      ],
      albums: [
        {
          _id: "5f1423c54ee369215e05625e",
          name: "Sample Album A"
        }
      ]
    },
    next: [
      {
        _id: "5f1422054ee369215e056256",
        name: "Cute",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056257",
        name: "Going Higher",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1ad1ef17e08f691bb74048",
        name: "Creative Minds",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          },
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      }
    ]
  },
  playlists: {
    loading: false,
    data: {
      "5f1ffca0faf3a97662868ac0": {
        name: "Test Playlist 1",
        songs: [
          {
            _id: "5f1422054ee369215e056257",
            name: "Going Higher",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1ad1ef17e08f691bb74048",
            name: "Creative Minds",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              },
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056255",
            name: "A New Beginning",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          }
        ]
      },
      "5f21389bc969de0689f10656": {
        name: "Test Playlist 2",
        songs: [
          {
            _id: "5f14218d4ee369215e05624f",
            name: "Hey",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          },
          {
            _id: "5f1422054ee369215e056256",
            name: "Cute",
            artists: [
              {
                _id: "5f14228b4ee369215e05625a",
                name: "Benjamin Tissot"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625e",
                name: "Sample Album A"
              }
            ]
          }
        ]
      },
      "5f213936c969de0689f10657": {
        name: "Test Playlist 3",
        songs: [
          {
            _id: "5f1422054ee369215e056258",
            name: "Ukulele",
            artists: [
              {
                _id: "5f14228b4ee369215e05625b",
                name: "BenSound"
              }
            ],
            albums: [
              {
                _id: "5f1423c54ee369215e05625f",
                name: "Sample Album B"
              }
            ]
          }
        ]
      }
    },
    error: null
  },
  songs: {
    loading: false,
    data: [
      {
        _id: "5f1422054ee369215e056255",
        name: "A New Beginning",
        artists: [
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625f",
            name: "Sample Album B"
          }
        ]
      },
      {
        _id: "5f1ad1ef17e08f691bb74048",
        name: "Creative Minds",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          },
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056256",
        name: "Cute",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056257",
        name: "Going Higher",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f14218d4ee369215e05624f",
        name: "Hey",
        artists: [
          {
            _id: "5f14228b4ee369215e05625a",
            name: "Benjamin Tissot"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625e",
            name: "Sample Album A"
          }
        ]
      },
      {
        _id: "5f1422054ee369215e056258",
        name: "Ukulele",
        artists: [
          {
            _id: "5f14228b4ee369215e05625b",
            name: "BenSound"
          }
        ],
        albums: [
          {
            _id: "5f1423c54ee369215e05625f",
            name: "Sample Album B"
          }
        ]
      }
    ],
    error: null
  }
};
