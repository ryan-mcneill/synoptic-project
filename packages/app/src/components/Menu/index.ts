import Accordion from "./Accordion";
import InfoTile from "./InfoTile";
import PlaylistTab from "./PlaylistTab";
import SearchTab from "./SearchTab";
import SongList from "./SongList";
import SongMenu from "./SongMenu";
import SongsTab from "./SongsTab";
import TabButton from "./TabButton";
import Tabs from "./Tabs";

export {
  Accordion,
  InfoTile,
  PlaylistTab,
  SearchTab,
  SongList,
  SongMenu,
  SongsTab,
  TabButton,
  Tabs
};
