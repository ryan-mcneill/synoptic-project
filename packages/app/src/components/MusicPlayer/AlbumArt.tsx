import React, { ReactElement } from "react";
import { connect, ConnectedProps } from "react-redux";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faMusic } from "@fortawesome/free-solid-svg-icons";
import { State } from "../../store/types";

const mapStateToProps = (state: State) => ({
  albumId: state.playlist.current?.albums[0]._id
});

const connector = connect(mapStateToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

const AlbumArt: React.FC<PropsFromRedux> = ({ albumId }): ReactElement => {
  return (
    <div
      className="album-art"
      style={{
        width: "80vw",
        height: "80vw",
        position: "absolute",
        top: "42vh",
        left: "50vw",
        transform: "translate(-50%, -50%)",
        backgroundColor: "#c0c0c0",
        boxShadow: "0px 0px 64px 4px rgba(0,0,0,0.2)"
      }}
    >
      {albumId ? (
        <img
          alt="Album art"
          src={`/api/album/art/${albumId}`}
          width="100%"
          height="100%"
        />
      ) : (
        <div
          style={{
            fontSize: "400px",
            position: "absolute",
            top: "50%",
            left: "48%",
            transform: "translate(-50%, -50%)"
          }}
        >
          <FontAwesomeIcon icon={faMusic} />
        </div>
      )}
    </div>
  );
};

export default connector(AlbumArt);
