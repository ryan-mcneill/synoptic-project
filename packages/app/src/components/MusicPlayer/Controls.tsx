import React, { ReactElement, useState } from "react";
import { connect, ConnectedProps } from "react-redux";
import classnames from "classnames";
import ReactAudioPlayer from "react-audio-player";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faPause,
  faPlay,
  faStepBackward,
  faStepForward
} from "@fortawesome/free-solid-svg-icons";
import { nextSong, prevSong, togglePlaying } from "../../store/actions";
import { State } from "../../store/types";

const mapDispatchToProps = (dispatch: any) => ({
  togglePlaying: (playing?: boolean) => dispatch(togglePlaying(playing)),
  nextSong: () => dispatch(nextSong()),
  prevSong: () => dispatch(prevSong())
});

const mapStateToProps = (state: State) => ({
  isDark: state.currentSong.background?.isDark,
  isPlaying: state.currentSong?.isPlaying,
  songId: state.playlist.current?._id,
  data: state.playlist.current
});

const connector = connect(mapStateToProps, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

const Controls: React.FC<PropsFromRedux> = ({
  data,
  isDark,
  isPlaying,
  songId,
  togglePlaying,
  nextSong,
  prevSong
}): ReactElement => {
  const [songLength, setSongLength] = useState(0);
  const [songTime, setSongTime] = useState(0);
  let audioPlayer: ReactAudioPlayer;

  const onSongLoad = () => {
    if (audioPlayer?.audioEl?.current) {
      if (isPlaying) audioPlayer.audioEl.current.play();
      setSongLength(audioPlayer.audioEl.current.duration);
    }
  };

  const formatTime = (time: number): string => {
    const tempDate = new Date(Math.round(time) * 1000);
    const minutes = tempDate.getMinutes();
    const seconds = tempDate.getSeconds();

    return (
      minutes.toString().padStart(2, "0") +
      ":" +
      seconds.toString().padStart(2, "0")
    );
  };

  const playStatusHandler = () => {
    if (data && audioPlayer?.audioEl?.current) {
      togglePlaying();
      audioPlayer.audioEl.current[isPlaying ? "pause" : "play"]();
    }
  };

  const nextHandler = () => {
    if (data) {
      nextSong();
      setSongTime(0);
    }
  };

  const prevHandler = () => {
    if (data) {
      prevSong();
      setSongTime(0);
    }
  };

  return (
    <div
      style={{
        height: "10vh",
        width: "90vw",
        position: "absolute",
        bottom: 0,
        padding: "0 5vw 5vh 5vw"
      }}
    >
      <ReactAudioPlayer
        ref={(element: ReactAudioPlayer) => (audioPlayer = element)}
        src={songId ? `/api/song/${songId}` : ""}
        listenInterval={1000}
        onListen={(e) => setSongTime(e)}
        onLoadedMetadata={() => onSongLoad()}
        onEnded={() => nextHandler()}
      />
      <div
        style={{
          display: "flex",
          alignItems: "center",
          justifyContent: "center"
        }}
      >
        <div style={{ fontSize: "48px" }} onClick={() => prevHandler()}>
          <FontAwesomeIcon icon={faStepBackward} />
        </div>
        <div
          style={{ fontSize: "96px", margin: "0 96px" }}
          onClick={playStatusHandler}
        >
          <FontAwesomeIcon icon={isPlaying ? faPause : faPlay} />
        </div>
        <div style={{ fontSize: "48px" }} onClick={() => nextHandler()}>
          <FontAwesomeIcon icon={faStepForward} />
        </div>
      </div>
      <div
        style={{
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          marginTop: "48px",
          fontSize: "32px"
        }}
      >
        <div>{formatTime(songTime)}</div>
        <input
          className={classnames({ dark: !isDark, light: isDark })}
          type="range"
          style={{ width: "100%", margin: "0 24px" }}
          min="0"
          max={songLength}
          defaultValue={songTime}
        />
        <div>{formatTime(songLength)}</div>
      </div>
    </div>
  );
};

export default connector(Controls);
