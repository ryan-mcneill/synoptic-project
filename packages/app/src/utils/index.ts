import { artistsToString } from "./artistsToString";
import { filterData } from "./filterData";
import { filterSongs } from "./filterSongs";
import { playlistsToArray } from "./playlistsToArray";

export { artistsToString, filterData, filterSongs, playlistsToArray };
