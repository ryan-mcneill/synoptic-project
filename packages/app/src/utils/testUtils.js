import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import reducers from "../store/reducers";

function render(
  ui,
  {
    initialState,
    store = createStore(reducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
