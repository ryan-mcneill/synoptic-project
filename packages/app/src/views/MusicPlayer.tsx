import React, { ReactElement } from "react";
import { connect, ConnectedProps } from "react-redux";
import { AlbumArt, Controls, NowPlaying } from "../components/MusicPlayer";
import { State } from "../store/types";

const mapStateToProps = (state: State) => ({
  albumsLoading: state.albums.loading,
  artistsLoading: state.artists.loading,
  isDark: state.currentSong.background?.isDark,
  backgroundColor: state.currentSong.background?.colour
});

const connector = connect(mapStateToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

const MusicPlayer: React.FC<PropsFromRedux> = ({
  albumsLoading,
  artistsLoading,
  isDark,
  backgroundColor
}): ReactElement =>
  !albumsLoading && !artistsLoading ? (
    <div
      style={{
        display: "block",
        margin: 0,
        position: "absolute",
        height: "100%",
        width: "100%",
        color: isDark ? "white" : "black",
        fontSize: "18px",
        backgroundColor
      }}
    >
      <NowPlaying />
      <AlbumArt />
      <Controls />
    </div>
  ) : (
    <></>
  );

export default connector(MusicPlayer);
