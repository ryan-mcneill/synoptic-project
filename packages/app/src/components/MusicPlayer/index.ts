import AlbumArt from "./AlbumArt";
import Controls from "./Controls";
import NowPlaying from "./NowPlaying";

export { AlbumArt, Controls, NowPlaying };
