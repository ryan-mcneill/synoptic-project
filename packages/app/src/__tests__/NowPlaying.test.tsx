import React from "react";
import { NowPlaying } from "../components/MusicPlayer";
import { render } from "../utils/testUtils";
import { mockState } from "../__mocks__/state";

describe("The NowPlaying component", () => {
  it("renders correctly with nothing loaded", () => {
    const { container } = render(<NowPlaying />, {
      initialState: { currentSong: {} }
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly after the data has loaded", () => {
    const { container } = render(<NowPlaying />, {
      initialState: mockState
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly when the data has but the artists and songName information isn't pulled through", () => {
    const initialState = { ...mockState };
    delete initialState.playlist.current.artists;
    delete initialState.playlist.current.name;
    const { container } = render(<NowPlaying />, {
      initialState
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
