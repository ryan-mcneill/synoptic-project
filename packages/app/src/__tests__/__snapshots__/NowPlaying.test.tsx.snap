// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The NowPlaying component renders correctly after the data has loaded 1`] = `
<div
  style="height: 5vh; width: 90vw; text-align: center; padding: 5vh 5vw 2vh 5vw;"
>
  <div
    style="font-size: 24px;"
  >
    NOW PLAYING
  </div>
  <div
    style="font-size: 48px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Hey - Benjamin Tissot
  </div>
</div>
`;

exports[`The NowPlaying component renders correctly when the data has but the artists and songName information isn't pulled through 1`] = `
<div
  style="height: 5vh; width: 90vw; text-align: center; padding: 5vh 5vw 2vh 5vw;"
>
  <div
    style="font-size: 24px;"
  >
    NOW PLAYING
  </div>
  <div
    style="font-size: 48px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Unknown - Unknown Artist
  </div>
</div>
`;

exports[`The NowPlaying component renders correctly with nothing loaded 1`] = `
<div
  style="height: 5vh; width: 90vw; text-align: center; padding: 5vh 5vw 2vh 5vw;"
>
  <div
    style="font-size: 24px;"
  >
    NOTHING PLAYING
  </div>
  <div
    style="font-size: 48px; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
  >
    Please make a song selection
  </div>
</div>
`;
